lib/ansible/module_utils/network/meraki/meraki.py

```diff
@@ -0,0 +1,287 @@
+
+    def is_org_valid(self, data, org_name=None, org_id=None):
+        ''' Checks whether a specific org exists and is duplicated '''
+        ''' If 0, doesn't exist. 1, exists and not duplicated. >1 duplicated '''
+        org_count = 0
+        if org_name is not None:
+            for o in data:
+                if o['name'] == org_name:
+                    org_count += 1
+        if org_id is not None:
+            for o in data:
+                if o['id'] == org_id:
+                    org_count += 1
+        return org_count
+
+    def get_org_id(self, org_name):
+        ''' Returns an organization id based on organization name, only if unique
+            If org_id is specified as parameter, return that instead of a lookup
+        '''
+        orgs = self.get_orgs()
+        if self.params['org_id'] is not None:
+            if self.is_org_valid(orgs, org_id=self.params['org_id']) is True:
+                return self.params['org_id']
+        org_count = self.is_org_valid(orgs, org_name=org_name)
+        if org_count == 0:
+            self.fail_json(msg='There are no organizations with the name {org_name}'.format(org_name=org_name))
+        if org_count > 1:
+            self.fail_json(msg='There are multiple organizations with the name {org_name}'.format(org_name=org_name))
+        elif org_count == 1:
+            for i in orgs:
+                if org_name == i['name']:
+                    # self.fail_json(msg=i['id'])
+                    return str(i['id'])
+
+    def get_net(self, org_name, net_name, data=None):
+        ''' Return network information '''
+        if not data:
+            org_id = self.get_org_id(org_name)
+            path = '/organizations/{org_id}/networks/{net_id}'.format(org_id=org_id, net_id=self.get_net_id(org_name=org_name, net_name=net_name, data=data))
+            return json.loads(self.request('GET', path))
+        else:
+            for n in data:
+                if n['name'] == net_name:
+                    return n
+
+    def get_net_id(self, org_name=None, net_name=None, data=None):
+        ''' Return network id from lookup or existing data '''
+        if not data:
+            self.fail_json(msg='Must implement lookup')
+        for n in data:
+            if n['name'] == net_name:
+                return n['id']
+        self.fail_json(msg='No network found with the name {0}'.format(net_name))
+
+    def construct_path(self, action, function=None, org_id=None, net_id=None, org_name=None):
+        built_path = None
+        if function is None:
+            built_path = self.url_catalog[action][self.function]
+        else:
+            self.function = function
+            built_path = self.url_catalog[action][function]
+        if org_name:
+            org_id = self.get_org_id(org_name)
+
+        ids = {'org_id': org_id,
+               'net_id': net_id,
+               }
+
+        built_path = built_path.format(**ids)
```

Comment:
As I indicated yesterday, here you could be doing:
```python
    built_path = build_path.format(org_id=org_id, net_id=net_id)
```
Rather than making a dict first, and then unfolding it for format().
For larger sets of variables you code is probably preferred.